src/software/firmware/srcs/activation.cpp

```diff
@@ -0,0 +1,50 @@
+/******************************************************************************
+ * @author Makers For Life
+ * @copyright Copyright (c) 2020 Makers For Life
+ * @file activation.h
+ * @brief Breathing activation related functions
+ *****************************************************************************/
+
+// INCLUDES ===================================================================
+
+// Associated header
+#include "../includes/activation.h"
+
+// External libraries
+
+// Internal
+#include "../includes/buzzer.h"
+
+const uint32_t SECOND_STOP_MAX_DELAY_MS = 5000;
+
+// INITIALISATION =============================================================
+
+ActivationController activationController;
+
+// FUNCTIONS ==================================================================
+
+ActivationController::ActivationController() : m_state(STOPPED), m_timeOfLastStopPushed(0) {}
+
+void ActivationController::onStartButton() { m_state = RUNNING; }
+
+void ActivationController::onStopButton() {
+    switch (m_state) {
+    case STOPPED:
+        break;
+
+    case RUNNING_READY_TO_STOP:
+        if (millis() - m_timeOfLastStopPushed < SECOND_STOP_MAX_DELAY_MS) {
+            m_state = STOPPED;
+            break;
+        }
+        // else fallback to default ON state handling
```

Comment:
Yes please

Reply:
Update comment as it's not an obvious behavior.